ranger/config.go

```diff
@@ -22,6 +22,10 @@ type Config[U RangerUser] struct {
 	// FS is the filesystem to find templates in for rendering them.
 	FS fs.FS
 
+	// MaintMode determines how to configure ranger on ranger.New.
+	// If true, it skips setting up a database connection and routes to a maintenance page.
+	MaintMode bool
```

Comment:
Went with a config value so that the caller could determine the logic on when to enable